File: src/dlp/ast/sentences/Return.ts

import {Expression} from "../expressions/Expression";
import {AbstractLocatable} from "../AbstractLocatable";
import {Sentence} from "./Sentence";

export class Assignment extends AbstractLocatable implements Sentence {

    returnValue: Expression;

    constructor( returnValue: Expression, line: number, column: number){
        super(line, column);
        this.returnValue = returnValue;
    }
}

File: src/ast/sentences/Return.java
package ast.sentences;

import ast.AbstractLocatable;
import ast.expressions.Expression;

public class Return extends AbstractLocatable implements Sentence {

    private Expression returnValue;

    public Return(Expression returnValue, int line, int column) {
        super(line, column);
        this.returnValue = returnValue;
    }

    public Expression getReturnValue() {
        return returnValue;
    }
}
